insecure/orchestrator/network.go

```diff
@@ -107,28 +107,46 @@ func (on *Network) stop() error {
 }
 
 // Observe is the inbound message handler of the orchestrator network.
-// Instead of dispatching their messages to the networking layer of Flow, the conduits of corrupt nodes
-// dispatch the outgoing messages to the orchestrator network by calling the InboundHandler method of it remotely.
+// Instead of dispatching their messages to the networking layer of Flow, the corrupt nodes
+// dispatch both the incoming (i.e., ingress) as well as the outgoing (i.e., egress)
+// messages to the orchestrator network by calling the InboundHandler method of it remotely.
 func (on *Network) Observe(message *insecure.Message) {
-	if err := on.processEgressMessage(message); err != nil {
-		on.logger.Fatal().Err(err).Msg("could not process message of corrupt node")
+	if message.Ingress != nil && message.Egress != nil {
```

Comment:
Do we need to add the 2nd error condition of having both ingress and egress messages set?

Reply:
Yes, per our conversation, exchanging an `insecure.Message` with both the ingress and egress set is an illegal state.